lib/vector-core/buffers/src/internal_events.rs

```diff
@@ -0,0 +1,42 @@
+use core_common::internal_event::InternalEvent;
+use metrics::counter;
+
+pub struct EventsReceived {
+    pub count: usize,
+    pub byte_size: usize,
+}
+
+impl InternalEvent for EventsReceived {
```

Comment:
These should be named something different than the generic `EventsReceived` and `EventsSent` in `src/internal_events/common.rs` to avoid name conflicts. I suggest `BufferEventsReceived` and `BufferEventsSent`.

Reply:
Gotcha, will rename. Out of curiosity, I'm assuming this naming conflict would occur if we decided to use, say, `buffer::internal_events::EventsReceived` and `common::EventsReceived` in the same module. Why not alias when importing instead?

Reply:
You are correct that the compiler will error if both are imported. However, there's a mental conflict as well, causing confusion over just which event is being emitted. To wit, when I first read this, I was going to ask about emitting the component-level event in the buffer, before realizing it was a completely different event. We do routinely violate this property (more than one type named `Error` or `Result` etc), so this isn't a hard-and-fast rule, just something to consider for types that already have a multi-word name.